lib/components/SelectionControl/checkbox/checkbox-group/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useState, useEffect } from "react";
+import clsx from "clsx";
+import "./checkbox-group.scss";
+import { Checkbox } from "../checkbox-single";
+import { CheckboxProps } from "../checkbox-single";
+
+interface Node extends Omit<CheckboxProps, "id" | "children"> {
+    id: string | number;
+}
+
+interface ParentNode extends Node {
+    children?: Node[];
+}
+
+interface CheckboxGroupProps {
+    className?: string;
+    checkboxGroupConfig: ParentNode[];
+}
+
+export const CheckboxGroup = ({
+    className,
+    checkboxGroupConfig,
+}: CheckboxGroupProps) => {
+    const [checkBoxItemsConfig, setCheckBoxItemConfig] =
+        useState(checkboxGroupConfig);
+
+    useEffect(() => {
+        const configCopy = structuredClone(checkBoxItemsConfig);
+        modifyConfig(configCopy);
+        setCheckBoxItemConfig(configCopy);
+    }, [checkboxGroupConfig]);
+
+    const updateParentState = (parent: ParentNode) => {
+        if (!parent.children) return;
+
+        const checkedChildrenSum = parent.children.reduce(
+            (acc, subItem) => acc + (subItem.checked ? 1 : 0),
+            0,
+        );
+        if (checkedChildrenSum === parent.children.length) {
+            parent.checked = true;
+            parent.indeterminate = false;
+        } else if (checkedChildrenSum === 0) {
+            parent.checked = false;
+            parent.indeterminate = false;
+        } else {
+            parent.indeterminate = true;
+        }
+    };
+
+    // Update 1) parent if children have field checked:true and it's not reflected in the Parent Node config part or 2) children if Parent is disabled
+    const modifyConfig = (config: ParentNode[]) => {
+        config.forEach((item) => {
+            if (item?.children) {
+                updateParentState(item);
+
+                if (item.disabled) {
+                    item.children.forEach(
+                        (subItem) => (subItem.disabled = true),
+                    );
+                }
+            }
+        });
+    };
+
+    const handleChildChange = (
```

Comment:
Just a thought , 
I see handleChildChange and handleParentChange shares a lot of similar functionality . Do you think we can create a single function for it.